### PR TITLE
Make relationship operators bidirectional

### PR DESCRIPTION
**Pull request by alexfurmenkov**

### CODE REVIEW

File: business_rules/operators.py [hunk 1]
```diff
@@ -657,10 +657,18 @@ def is_unique_relationship(self, other_value):
...
+        if comparator_is_to_the_right_of_target:
+            grouped_dict = self.value.groupby([comparator])[target].apply(set).to_dict()
+            results = np.where(
+                vectorized_len(vectorized_get_dict_key(grouped_dict, self.value[comparator])) <= 1, True, False
+            )
+        else:
+            grouped_dict = self.value.groupby([target])[comparator].apply(set).to_dict()
+            results = np.where(
+                vectorized_len(vectorized_get_dict_key(grouped_dict, self.value[target])) <= 1, True, False
+            )
```
Comment (alexfurmenkov) [inline business_rules/operators.py:671]: The solution looks a bit ugly but it works. Will try to implement a more elegant way.

File: business_rules/operators.py [hunk 2]
```diff
@@ -650,22 +650,54 @@ def is_unique_set(self, other_value):
...
-        grouped_dict = self.value.groupby([comparator])[target].apply(set).to_dict()
-        results = np.where(
-            vectorized_len(vectorized_get_dict_key(grouped_dict, self.value[comparator])) <= 1, True, False
-        )
-        return pd.Series(results)
+        df_without_duplicates: pd.DataFrame = self.value.drop_duplicates()  # remove repeating rows
+        # we need to check if ANY of the columns (target or comparator) is duplicated
+        duplicated_comparator: pd.Series = df_without_duplicates[comparator].duplicated(keep=False)
+        duplicated_target: pd.Series = df_without_duplicates[target].duplicated(keep=False)
+        if duplicated_comparator.any() and not duplicated_target.any():
```
Comment (nhaydel) [inline business_rules/operators.py:683]: Is it possible for more than one of these cases to be triggered for the same dataset?

Comment (alexfurmenkov) [reply business_rules/operators.py:683]: Yes. This case is handled by `elif duplicated_target.any() and duplicated_comparator.any():` condition

Comment (alexfurmenkov) [reply business_rules/operators.py]: **UPDATE:** I have refactored the code to make it a bit more elegant and the condition `elif duplicated_target.any() and duplicated_comparator.any():` doesn't exist any more but anyway the case you had mentioned is handled.

**Pull request merged**